core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -1168,17 +1175,18 @@ private ExprValue exprYear(ExprValue date) {
     return new ExprIntegerValue(date.dateValue().getYear());
   }
 
+  private LocalDateTime formatNow(Clock clock) {
+    return formatNow(clock, 0);
+  }
+
   /**
    * Prepare LocalDateTime value. Truncate fractional second part according to the argument.
    * @param fsp argument is given to specify a fractional seconds precision from 0 to 6,
    *            the return value includes a fractional seconds part of that many digits.
    * @return LocalDateTime object.
    */
-  private LocalDateTime formatNow(@Nullable Integer fsp) {
-    var res = LocalDateTime.now();
-    if (fsp == null) {
-      fsp = 0;
-    }
+  private LocalDateTime formatNow(Clock clock,  Integer fsp) {
+    var res = LocalDateTime.now(clock);
```

Comment:
do we care about the `fsp == null` case?

Reply:
It's now handled by
https://github.com/Bit-Quill/opensearch-project-sql/blob/7dcbed1ca6163b3c85e4f235f744363099dc7735/core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java#L1178-L1180